### DISCUSSION TITLE
Enforce class annotation at call site?

### DISCUSSION THREAD

**Discussion by carterhudson** [detekt/detekt#6968] [Q&A] [answered]
Is there a way to write a custom rule to act like `StringRes` does?
![Screenshot 2024-02-12 at 3 18 09 PM](https://github.com/detekt/detekt/assets/2574098/47ba8cd6-0cc7-4fa0-9a8d-429f2f109425)

Basically I'm trying to do:
``` kotlin
@Retention(AnnotationRetention.RUNTIME)
@Target(
    AnnotationTarget.CLASS,
    AnnotationTarget.VALUE_PARAMETER
)
annotation class SomeAnnotation

@SomeAnnotation
class Foo

fun someFun(@SomeAnnotation someParam: Any) {
    // ...
}

fun main() {
    someFun(2) // <- should underline and note that it should be a type annotated with @SomeAnnotation

    someFun(Foo()) // <- works fine
}
```

**Comment by BraisGabin** [comment, marked as answer]
Yes, this is doable. You will need to use the `ContextBinding` but that should be perfectly doable.

**Reply by carterhudson** [reply]
Any relevant Rule examples you can point to? I'm not super familiar with the Kotlin PSI API, so any hints are appreciated.
